snuba/query/logical.py

```diff
@@ -396,10 +398,10 @@ def validate_aliases(self) -> bool:
         declared_symbols |= {c.flattened for c in self.get_data_source().get_columns()}
         return not referenced_symbols - declared_symbols
 
-    def set_entity_name(self, entity_name: str) -> None:
-        assert self.__entity_name is None
-        self.__entity_name = entity_name
+    def set_entity(self, entity: Entity) -> None:
```

Comment:
Can't this take any data source? Is there a reason we are locking this to Entity?

Reply:
It is in a further step. 
The query class will become Generic with respect to the type of FROM clause it allows so we can ensure the entity query processor will only receive Query[Entity], the physical query processors will receive Query[RelationalSource] and the composed query can have something more elaborate like joins or subqueries.

I would not have touched the query class at all in his PR if the entity name had not already been added, butt since it was already there, I could use it to validate everything works.